geoipupdate/geoip.py

```diff
@@ -78,23 +82,24 @@ def download_file(current_directory):
     downloaded_file = os.path.join(current_directory + GEOIP_PATH, MAXMIND_FILENAME)
     existing_file = os.path.join(current_directory + GEOIP_PATH, LOCAL_FILENAME)
     urllib.request.urlretrieve(
-        urllib.parse.urljoin(MAXMIND_URL, MAXMIND_FILENAME),
+        MAXMIND_URL,
         downloaded_file
     )
     logging.info("Downloading completed geoip2 country database  ")
-    with gzip.open(downloaded_file, 'rb') as outfile:
-
-        md5_url = urllib.parse.urljoin(
-            MAXMIND_URL,
-            MAXMIND_FILENAME.split('.', 1)[0] + '.md5'
-        )
-        if not match_md5(outfile, md5_url):
-            logging.info("Downloaded file hash did't matched. Aborting the operation")
-            try:
-                os.remove(downloaded_file)
-            finally:
-                raise ValueError(
-                    'md5 of %s doesn\'t match the signature.' % downloaded_file
-                )
-        write(outfile, existing_file)
-    os.remove(downloaded_file)
+
+    if match_sha256(downloaded_file, MAXMIND_SHA256_URL):
+
+        with tarfile.open(downloaded_file, 'r:gz') as outfile:
+            for member in outfile.getmembers():
+                if LOCAL_FILENAME in member.name:
+                    _file = outfile.extractfile(member)
+                    write(_file, existing_file)
+        os.remove(downloaded_file)
+    else:
+        logging.info("Downloaded file hash did't matched. Aborting the operation")
+        try:
+            os.remove(downloaded_file)
+        finally:
+            raise ValueError(
+                'md5 of %s doesn\'t match the signature.' % downloaded_file
```

Comment:
Change the log to to output **sha256.** . Plus also change the docstring https://github.com/edx/ecommerce-scripts/blob/master/geoipupdate/geoip.py#L64